hw/snitch_cluster/src/snitch_cluster_peripheral/snitch_cluster_peripheral_reg.rdl

```diff
@@ -0,0 +1,242 @@
+        };
+        ICACHE_STALL = 30 {
+            desc = "Incremented for instruction cache stalls.
+                    _This is a hart-local signal_";
+        };
+    };
+
+    reg PERF_CNT_SEL {
+        desc = "Select the metric that is tracked for each performance counter.";
+        field {
+            name = "HART";
+            desc = "Select from which hart in the cluster, starting from `0`,
+                    the event should be counted. For each performance counter
+                    the cores can be selected individually. If a hart greater
+                    than the cluster's total hart size is selected the selection
+                    will wrap and the hart corresponding to `hart_select % total_harts_in_cluster`
+                    will be selected.";
+            hw = rw;
+            sw = rw;
+        } HART[15:0];
+        field {
+            name = "METRIC";
+            desc = "Select the metric that is tracked for each performance counter";
+            hw = rw;
+            sw = rw;
+            encode = PERF_METRIC;
+        } METRIC[31:16];
+    };
+
+    reg PERF_CNT {
+        desc = "Performance counter. Set corresponding PERF_CNT_SEL register depending on what
+                performance metric and hart you would like to track.";
+        field {
+            desc = "Performance counter";
+            hw = rw;
+            sw = rw;
+            onwrite = wclr;
+        } PERF_COUNTER[47:0];
+    };
+
+    reg SCRATCH {
+        desc = "Scratch registers. Used in the bootrom for various purposes.";
+        field {
+            desc = "Scratch register";
+            hw = na;
+            sw = rw;
+        } SCRATCH[31:0];
+    };
+
+    reg CL_CLINT_SET {
+        desc = "Set bits in the cluster-local CLINT. Writing a 1 at location i sets the cluster-local interrupt
+                of hart i, where i is relative to the first hart in the cluster, ignoring the cluster base hart ID.";
+        field {
+            desc = "Set cluster-local interrupt of hart i";
+            sw = w;
+            hw = r;
+        } CL_CLINT_SET[31:0];
+    };
+
+    reg CL_CLINT_CLEAR {
+        desc = "Clear bits in the cluster-local CLINT. Writing a 1 at location i clears the cluster-local interrupt
+                of hart i, where i is relative to the first hart in the cluster, ignoring the cluster base hart ID.";
+        field {
+            desc = "Set cluster-local interrupt of hart i";
+            sw = w;
+            hw = r;
+        } CL_CLINT_CLEAR[31:0];
+    };
+
+    reg ICACHE_PREFETCH_ENABLE {
+        desc = "Controls prefetching of the instruction cache.";
+        field {
+            desc = "Enable instruction prefetching.";
+            hw = r;
+            sw = w;
+        } ICACHE_PREFETCH_ENABLE[0:0];
+    };
+
+             PERF_CNT_EN    PERF_CNT_EN[NumPerfCounters]   @0x0   += 0x8;
+    external PERF_CNT_SEL   PERF_CNT_SEL[NumPerfCounters]  @0x100 += 0x8;
```

Comment:
Is it possible to just align the address to the previous regfile? Not a big problem, just curious

Reply:
I see👍 I don't think the actual address is interesting in the end as long as we reference the generated files. I think the `compact` statement just means that multiple fields can be bundled into a 32-bit register (to save some address space I guess). But I don't think we actually require it.